components/docs-chef-io/content/habitat/origin-keys.md

```diff
@@ -0,0 +1,190 @@
++++
+title = "Origin Keys"
+
+date = 2020-10-12T13:59:46-07:00
+draft = false
+
+[menu]
+  [menu.habitat]
+    title = "Origin Keys"
+    identifier = "habitat/origins/origin-keys Origin Keys"
+    parent = "habitat/origins"
+    weight = 20
++++
+
+[\[edit on GitHub\]](https://github.com/habitat-sh/habitat/blob/master/components/docs-chef-io/content/habitat/origin-keys.md)
+
+Prerequisites:
+
+- [Get Chef Habitat]({{< relref "install-habitat.md" >}})
+- [Create a Chef Habitat Builder account]({{< relref "builder-account#builder-account" >}})
+- [Generate a personal access token]({{< relref "builder-profile#create-a-personal-access-token" >}})
+- [Create an origin]({{< relref "origins" >}}) or join an origin by [invitation]({{< relref "origin-rbac#manage-origin-membership-with-hab-origin-invitations" >}})
+
+When you create an origin, Chef Habitat Builder automatically generates _origin keys_.
+Origin key cryptography is asymmetric: it has a public origin key that you can distribute freely, and a private origin key that you should distribute only to users of the origin.
+All Chef Habitat Builder users with access to the origin can view the origin public key revisions in the origin key tab (Builder > Origin > Keys) and download the origin public key, but only users with the origin 'administrator' or 'owner' roles can view or download the origin private key, or change the origin key pair.
+
+| Keys Actions | Read-Only | Member | Maintainer | Administrator | Owner |
+|---------|-------|-------|-------|-------|-------|
+| View keys | Y | Y | Y | Y | Y |
+| Add/Update/Delete keys | N | N | N | Y | Y |
+
+Chef Habitat uses origin keys:
+
+- When you build an artifact in your local environment, Chef Habitat signs the artifact with a public key
+- When you upload an artifact to Chef Habitat Builder or Builder on-prem, Chef Habitat verifies that the artifact was signed with its public key
+- When you install an artifact on a Chef Habitat Supervisor, Chef Habitat uses the public origin key to authorize the artifact's installation; Chef Habitat only installs artifacts for which it has the public origin key
+- When you download an artifact to your local environment, Chef Habitat uses the public origin key to verify the artifact's integrity before it starts the installation
+
+Chef Habitat Builder origin key names follow the format:
+
+```hab
+<origin>-<datetime>.pub (public key)
+<origin>-<datetime>.sig.key (private key, also called a "signing key")
+```
+
+For example, in:
+
+```hab
+testorigin-20190416223046.pub
+testorigin-20190416223046.sig.key
+```
+
+- "testorigin" is the origin's name
+- "20190416223046" is the date and time of the key's creation, which was 2019-04-16 22:30:46.
+- `.pub` is the file extension for the public key
+- `.sig.key` is the file extension for the private key, which is also called a "signing key"
```

Comment:
While I don't think there's really a semantic difference, the code refers to all the "private" keys as "secret". If we could be consistent across code and documentation, I think that would be nice.

Reply:
Yes, but the common term is "private key" not "secret key", so I would want to change the code. :fightme:

Reply:
Counterpoint: the crypto library we're using calls them secret keys 😂 

https://github.com/habitat-sh/habitat/blob/e3d33d0bcc0d4d0585b25061941a02f489531478/components/core/src/crypto/keys/signing.rs#L15-L16

![tenor-114036778](https://user-images.githubusercontent.com/207178/96140540-0da68200-0ece-11eb-8eba-66291a114a02.gif)